vimPlugins.avante-nvim: 2024-10-18 -> 2024-11-04

ttrei opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

ttrei committed:
vimPlugins.avante-nvim: 2024-10-18 -> 2024-11-04

GaetanLepage commented on pkgs/applications/editors/vim/plugins/overrides.nix:
```diff
@@ -163,14 +163,77 @@ in
+          [
+            openssl
+          ]
+          ++ lib.optionals stdenv.hostPlatform.isDarwin [
+            darwin.apple_sdk.frameworks.Security
+          ];
```
You can probably get rid of this now as the darwin refactor has landed in `master`.

khaneliman replied:
Yeah, if we're going to touch all of the code for this plugin, could you remove this optional? It will automatically get all apple-sdk frameworks in the stdenv, now. I'm not familiar enough with their dependencies to know if it could benefit from a higher level SDK. But, I do see they are targeting macos-latest in their CI runner.

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 353494`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.avante-nvim</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.avante-nvim</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>vimPluginsUpdater</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>vimPlugins.avante-nvim</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>vimPluginsUpdater</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>vimPlugins.avante-nvim</li>
  </ul>
</details>

GaetanLepage reviewed (approved):
LGTM !
(failures on darwin are unrelated)

khaneliman commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 353494`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.avante-nvim</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.avante-nvim</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>vimPluginsUpdater</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>vimPlugins.avante-nvim</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>vimPluginsUpdater</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>vimPlugins.avante-nvim</li>
  </ul>
</details>

aarnphm commented on pkgs/applications/editors/vim/plugins/overrides.nix:
```diff
@@ -163,14 +163,71 @@ in
-      plenary-nvim
-    ];
-  };
+  avante-nvim =
```
quick question, why do we derive away from other plugins?

afaict every other plugin is using overrideAttrs? Wouldn't this make it easier to maintain with what we have like before?

khaneliman replied:
This plugin isn't part of the generated.nix like the others. I'm not familiar with the history of this plugin and why it's doing it this way... but might be nice to just setup a derivation for `avante-nvim-lib` and then move `avante.nvim` to behave similar to the others and use overrides.nix to handle just the postInstall and dependencies

khaneliman replied:
If the lib is built from the same source as the vimPlugin then you could just have the main plugin source get updated in generated.nix and the overrideAttrs in here could do the `avante-nvim-lib` build for the postInstall.

aarnphm replied:
quick context: we have a few bindings in rust, therefore we probably need a `avante-nvim-lib` derivation (similar to what you suggest here)

otherwise I think your suggestion makes sense

GaetanLepage replied:
I am the one who packaged this plugin if I remember well.
First, the plugin has semantic releases, which encourages for using those instead of the latest git commit.
Also, as it needed some more work in `overrides.nix`, it was logical to have it completely packaged here.

There are examples of plugins which build some rust binaries while having their source updated using the script (part of `generated.nix`). The issue is that if someone updates the `vimPlugins` package set without paying attention, the `carhoHash` can become outdated and thus the plugin will not build anymore.

GaetanLepage replied:
> First, the plugin has semantic releases, which encourages for using those instead of the latest git commit.

Well, I made a mistake and thought about another plugin. This plugin has no semantic releases.
It would then make more sense to have its source being updated as part of the `generated.nix` process.

@ttrei, could you go ahead and modify `vim-plugin-names`, `generated.nix` and `overrides.nix` accordingly ?

GaetanLepage replied:
(Also, I am not the one who has packaged it in nixpkgs, but I added the nixvim module).

GaetanLepage replied:
> @ttrei, could you go ahead and modify `vim-plugin-names`, `generated.nix` and `overrides.nix` accordingly ?

Here is a proposition: https://github.com/NixOS/nixpkgs/commit/4a7dfbd96f53a57c3f1fc5875911a9fc8c562125
If everything looks good to you, just pick the changes and update your branch.

ttrei replied:
@GaetanLepage, you removed the meta.maintainers and other attributes, is this intentional?

GaetanLepage replied:
Indeed, this is a mistake.
Fixed in https://github.com/GaetanLepage/nixpkgs/commit/53a3522af3e0b4fb43c5a57e9ff41a96047a75f0.

ttrei replied:
thanks! included your changes in the pr

do i understand correctly that now the plugin version will get updated during mass plugin updates?

GaetanLepage replied:
> do i understand correctly that now the plugin version will get updated during mass plugin updates?

Yes. When running the script though, `cargoHash` will have to be updated manually.
This would be surely caught when running `nixpkgs-review` so I am not too worried.

PerchunPak replied:
> cargoHash will have to be updated manually.

This would be really annoying, maybe return to the previous method but just add an update script?

GaetanLepage reviewed (approved):
LGTM thanks !

GaetanLepage commented:
Well, you have conflicts in `vim-plugin-names`.

ttrei commented:
rebased onto latest master to resolve conflicts

also updated to latest upstream commit - there was a larger feature merged a few moments ago
i suggest waiting one day before merging into nixpkgs, to catch any obvious bugs

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 353494`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.avante-nvim</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.avante-nvim</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.avante-nvim</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.avante-nvim</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

GaetanLepage commented:
> i suggest waiting one day before merging into nixpkgs, to catch any obvious bugs

Not really needed. Just mark it as open for review.
If ofborg is successfull we can just merge it.

aarnphm reviewed (approved):
LGTM
